Add a OMERO.table visualisation example

fmeyenhofer opened the pull request:
- New section with the mechanisms and screenshot
- Link to the python examples.

As discussed with @will-moore in 
https://forum.image.sc/t/omero-tables-display-on-images-link-questions/87870

The updated code examples in #2355 are part of this same issue.

This is the second, revised pull-request replacing  #2354.

fmeyenhofer committed:
Add a OMERO.table visualisation example

- New section with the mechanisms and screenshot
- Link to the python examples.

will-moore commented on omero/developers/Tables.rst:
```diff
@@ -295,13 +295,49 @@ These are only an introduction to using OMERO.tables and do not show its full
+       |- Image 2 <- (Table panel > link, row values)
+       |- ...
+
+The rows of the table will be rendered for each image of the dataset. The screenshot bellow shows the right-hand side panel of a selected Image in OMERO.web.
```
typo: "below".
Also nicer to use "Image" and "Dataset" instead of "image" and "dataset". Same with sentence above.

will-moore commented on omero/developers/Tables.rst:
```diff
@@ -295,13 +295,49 @@ These are only an introduction to using OMERO.tables and do not show its full
+Data viewing in OMERO.web
+-------------------------
+
+OMERO.web can display table data in a separate tab (attachment link) or in the image context, given the following setup:
```
Tables are displayed in a similar way in the Insight client too.

fmeyenhofer replied:
Do you refer to the ROI-Tool?
Could this be a separate section? 
(Viewing the attachment from a Dataset or Project in OMERO.insight did not look very readable to me).

will-moore replied:
Apologies - you're right. Insight implemented this for Screen/Plate/Well but never did it for Project/Dataset/Image.

fmeyenhofer committed:
Update Tables.rst

Use capitals for Image and Dataset

will-moore commented:
Just reading the built pages at https://omero--2357.org.readthedocs.build/en/2357/developers/Tables.html
This says you have to attach the Table to a Dataset, but you can also attach it to a Project, Screen or Plate that contains the Images.

fmeyenhofer commented:
Yes, I was thinking about this in the context of OMERO.parade. 
@Rdornier wrote about in his wiki https://wiki-biop.epfl.ch/en/data-management/omero/parade. He mentions possible conflicts about OMERO.tables in OMERO.parade (but this is another topic). 

I did not mention the screen-plate-well use case - since I do not have any experience with that yet only just became aware of it. 

This should definitely be mentioned.

fmeyenhofer committed:
Update Tables.rst

Fix typo

will-moore commented:
Nice wiki pages on OMERO.parade and other OMERO usage!
@fmeyenhofer So you're going to mention that this works with Project, Screen & Plate?

fmeyenhofer committed:
Mention all data objects in the OMERO.table visualisation example

- Mention Prject, Screen, Plate and well objects.
- Add link to the Screen Plate Well data model
- Add link to OMERO.parade
- Refine text

fmeyenhofer commented:
Done

will-moore reviewed (approved):
Looks good, thanks

will-moore commented:
@fmeyenhofer could I ask you to please read and fill-out a CLA form as described at https://ome-contributing.readthedocs.io/en/latest/cla.html - We need this before we can accept contributions to this repo. Many thanks
